test/replication-luatest/gh_7318_box_wait_limbo_acked_test.lua

```diff
@@ -0,0 +1,545 @@
+        return f:id()
+    end)
+    server_wait_wal_is_blocked(cg.replica)
+
+    cg.replica_proxy:pause()
+
+    t.helpers.retrying({timeout = wait_timeout}, function()
+        cg.master:exec(function()
+            local status = box.info.replication[2].upstream.status
+            t.assert(status ~= 'follow')
+        end)
+    end)
+
+    cg.master:exec(function()
+        require('fiber').create(function() box.space.test:insert{1} end)
+    end)
+    server_wait_synchro_queue_len_is_equal(cg.replica, 1)
+
+    local repl = cg.replica:exec(function()
+        local repl = table.copy(box.cfg.replication)
+        box.cfg{replication={repl[1]}}
+        return repl
+    end)
+
+    cg.master:exec(function()
+        box.cfg{replication_synchro_quorum=1}
+    end)
+    server_wait_synchro_queue_len_is_equal(cg.master, 0)
+    cg.master:exec(function()
+        box.cfg{replication_synchro_quorum=2}
+    end)
+
+    local csw = get_fiber_csw(cg.replica, f)
+    local ff = require('fiber').create(function()
+        cg.replica:exec(function(f)
+            box.error.injection.set('ERRINJ_WAL_DELAY', false)
+            local _, ok, err = require('fiber').find(f):join()
+            t.assert(ok)
+            t.assert(not err)
+        end, {f})
+    end)
+
+    ff:set_joinable(true)
+    server_wait_fiber_csw_ge_than(cg.replica, f, csw + 1)
+
+    cg.replica:exec(function(repl)
+        box.cfg{replication=repl}
+    end, {repl})
+
+    server_wait_synchro_queue_len_is_equal(cg.replica, 0)
+    cg.replica_proxy:resume()
+    local ok, err = ff:join()
+    t.assert(ok)
+    t.assert(not err)
+end
+
+g.test_quorum_less_replication_synchro_quorum = function(cg)
+    local f = cg.replica:exec(function()
+        box.error.injection.set('ERRINJ_WAL_DELAY_COUNTDOWN', 0)
+        local f = require('fiber').create(function()
+            return pcall(box.ctl.promote)
+        end)
+        f:set_joinable(true)
+        return f:id()
+    end)
+    server_wait_wal_is_blocked(cg.replica)
+    cg.master:exec(function()
+        require('fiber').create(function() box.space.test:insert{1} end)
+    end)
+    cg.replica:exec(function()
+        box.cfg{replication_synchro_quorum=1}
+    end)
+    server_wait_synchro_queue_len_is_equal(cg.replica, 1)
+    local csw = get_fiber_csw(cg.replica, f)
+    cg.replica:exec(function()
+        box.error.injection.set('ERRINJ_WAL_DELAY_COUNTDOWN', 0)
+        box.error.injection.set('ERRINJ_WAL_DELAY', false)
+    end)
+    server_wait_fiber_csw_ge_than(cg.replica, f, csw + 1)
+    cg.replica:exec(function()
+        box.cfg{replication_synchro_quorum=2}
+    end)
+    csw = get_fiber_csw(cg.replica, f)
+    cg.replica:exec(function()
+        box.error.injection.set('ERRINJ_WAL_DELAY', false)
+    end)
+    --we need to be sure we entered the box_wait_quorum call
+    server_wait_fiber_csw_ge_than(cg.replica, f, csw + 1)
+    cg.replica:exec(function(f)
+        box.cfg{replication_synchro_quorum=1}
```

Comment:
Is it necessary to reset the quorum here?

Reply:
It is unnecessary.